core/src/world/place/location/geographical/beach.rs

```diff
@@ -0,0 +1,182 @@
+        0 => land_animal(rng),
+        1..=2 => coastal_animal(rng),
+        3 => enemy(rng),
+        4 => food(rng),
+        5 => profession(rng),
+        6 => symbol(rng),
+        7 => gem(rng),
+        _ => unreachable!(),
+    }
+}
+
+fn pluralize(word: &str) -> (&str, &str) {
+    match word {
+        "Goose" => ("geese", ""),
+        "Key" => ("key", "s"),
+        "Beef" | "Carp" | "Cod" | "Deer" | "Perch" | "Potatoes" | "Sheep" | "Squid" => (word, ""),
+        s if s.ends_with('f') => (&word[..(word.len() - 1)], "ves"),
+        s if s.ends_with("ey") => (&word[..(word.len() - 2)], "ies"),
+        s if s.ends_with('y') => (&word[..(word.len() - 1)], "ies"),
+        s if s.ends_with(&['s', 'x'][..]) => (word, "es"),
+        _ => (word, "s"),
+    }
+}
+
+fn adjective(rng: &mut impl Rng) -> &'static str {
+    #[rustfmt::skip]
+    const ADJECTIVES: &[&str] = &[
+        "Blue", "Bronze", "Brown", "Burgundy", "Driven", "Enchanted", "Gold", "Green", "Grey",
+        "Grouchy", "Hallowed", "Happy", "Hidden", "Hungry", "Jovial", "Lone", "Lost", "Lucky",
+        "Merry", "Moody", "Morose", "Orange", "Purple", "Red", "Silent", "Silver", "Thirsty",
+        "Wasted", "Wild",
+    ];
+    ADJECTIVES[rng.gen_range(0..ADJECTIVES.len())]
+}
+
+fn beach_synonym(rng: &mut impl Rng) -> &'static str {
+    #[rustfmt::skip]
+    const BEACH_SYNONYMS: &[&str] = &[
+        "Bank", "Beach", "Berm", "Coast", "Cove", "Embankment", "Landing", "Point", "Sands",
+        "Shore", "Shoreline", "Strand", "Waterfront",
+    ];
+    BEACH_SYNONYMS[rng.gen_range(0..BEACH_SYNONYMS.len())]
+}
+
+fn land_animal(rng: &mut impl Rng) -> &'static str {
+    #[rustfmt::skip]
+    const LAND_ANIMALS: &[&str] = &[
+        "Antelope", "Ape", "Baboon", "Badger", "Bat", "Bear", "Beaver", "Bee", "Beetle", "Boar",
+        "Camel", "Cat", "Cow", "Deer", "Dog", "Donkey", "Dove", "Dragonfly", "Duck", "Eagle",
+        "Elephant", "Elk", "Ermine", "Fox", "Frog", "Goat", "Goose", "Hare", "Hart", "Hawk",
+        "Hedgehog", "Heron", "Herring", "Horse", "Hound", "Hyena", "Jackal", "Lamb", "Leopard",
+        "Lion", "Magpie", "Mole", "Owl", "Panther", "Peacock", "Phoenix", "Pony", "Porcupine",
+        "Rabbit", "Ram", "Rat", "Raven", "Salamander", "Scorpion", "Sheep", "Snake", "Spider",
+        "Squid", "Squirrel", "Stag", "Stoat", "Stork", "Swan", "Tiger", "Toad", "Tortoise",
+        "Turkey", "Turtle", "Unicorn", "Vulture", "Weasel", "Wolf",
+    ];
+    LAND_ANIMALS[rng.gen_range(0..LAND_ANIMALS.len())]
+}
+
+fn coastal_animal(rng: &mut impl Rng) -> &'static str {
+    #[rustfmt::skip]
+    const COASTAL_ANIMALS: &[&str] = &[
+        "Cormorant", "Crab", "Dolphin", "Herring", "Mermaid", "Octopus", "Osprey", "Otter",
+        "Pelican", "Perch", "Salmon", "Seagull", "Seal", "Shark", "Starfish", "Squid", "Whale",
+        "Whelk",
+    ];
+    COASTAL_ANIMALS[rng.gen_range(0..COASTAL_ANIMALS.len())]
+}
+
+fn enemy(rng: &mut impl Rng) -> &'static str {
```

Comment:
I think it's worth DRYing out the vocabulary with `inn` and `shrine` at this point, although doing so will presumably require updating the tests for all three. I propose adding a new mod `crate::world::place::vocabulary` and putting the various general vocabulary there.

Doing so should definitely simplify adding future generators.

Reply:
Correction: `crate::world::vocabulary`. Seems a bit silly to tie it too closely to `place`. I can see a case for having an NPC named Dera Jacasshield the Lost.

Reply:
Yeah that sounds good to me. I can start an issue/branch for this if you like?

Reply:
Sure, seems reasonable to do it separately since you've already got the vocab implemented here.